Sources/LeafKit/LeafSyntax/LeafSyntax.swift

```diff
@@ -18,29 +18,133 @@ public enum ConditionalSyntax {
     case `if`([ParameterDeclaration])
     case `elseif`([ParameterDeclaration])
     case `else`
+    
+    func imports() -> Set<String> {
+        switch self {
+            case .if(let pDA), .elseif(let pDA):
+                var imports = Set<String>()
```

Comment:
Similar to above, this could be one of:
- `return pDA.reduce(Set<String>()) { $0.formUnion($1.imports()) }`
- `return Set(pDA.lazy.flatMap { $0.imports() })`

Reply:
Leaving as-is for clarity - side note btw - given that these functions are only called during template resolution (never during data serialization), I'm not super concerned about any minor performance differences and prefer clear behavior :)

Reply:
As above, the `into:` label is missing in my code. But as long as it’s not performance critical, leaving it as is is also fine. 🙂

Reply:
IIRC I tried it with reduce originally but could swear I got compiler issues with it having trouble on type inference, and it ended up being even more awkward than the foreach version

Reply:
As long as you qualify the Element Parameter of Set, the closure should be typed enough to infer all remaining types. But yea, sometimes weird things happen with type inference. 😀

Reply:
Yeah - should clarify, `reduce(into` was fine for *compiling* but was incapable of inferring $0 and $1 in the closure, and I found that obnoxious :)

Reply:
@tdotclare Hehe, the "while you're typing" problem is because there are two parameters in the closure ($0 and $1). As long as you only use one, the compiler is unsure whether you're going to refer to the second one as `$0.1` or `$1`.
For getting auto-completion while typing, what works is this:
```swift
return pDA.reduce(into: Set<String>()) {
    let imports = $1.imports()
    $0.formUnion(imports)
}
```

I know it's dumb. They should actually ban the `$0.1` access to non-tuple parameters in closures in which case the compiler would know the type of `$0` while you're typing.